Add {SortedSet,RetrySet}.range_with_score/3

keyan opened the pull request:
This additional method allows for listing jobs by their precedence within a Redis set. This can be useful for determining when a job will next be attempted or when a dead job was declared dead, among other possible uses.

keyan commented:
This method is needed for https://github.com/edgurgel/verk_web/issues/22

edgurgel commented on lib/verk/sorted_set.ex:
```diff
@@ -68,6 +68,36 @@ defmodule Verk.SortedSet do
+          jobs
+          |> Enum.take_every(2)
+          |> Enum.map(fn job -> Job.decode!(job) end)
+        {:ok, Enum.zip(decoded_jobs, scores)}
```
I wonder if you could do something like:

```elixir
 jobs |> Enum.chunk(2) |> Enum.into([], fn [job, score] -> {Job.decode!(job) ,score} end)
```

keyan replied:
This is great! I was trying to find something like this, but didn't notice that method :)

edgurgel commented on lib/verk/retry_set.ex:
```diff
@@ -86,6 +86,22 @@ defmodule Verk.RetrySet do
+  List jobs from `start` to `stop` including their scores
+  """
+  @spec range_with_score(integer, integer, GenServer.server) :: {:ok, [{Verk.Job.T, integer}]} | {:error, Redix.Error.t}
+  def range_with_score(start \\ 0, stop \\ -1, redis \\ Verk.Redis) do
```
We also need to add tests to the `RetrySet` counterpart

keyan commented:
@edgurgel, ready for another look. I have addressed your suggestions.

If you think it is good to merge just let me know, as I'd rather rebase these commits to clean up the git history before merging.

edgurgel commented:
LGTM! If you could rebase that would be great 👍

keyan commented:
@edgurgel, rebased, ready to go! Thanks!

edgurgel commented:
Thank you 🎉 I still need to release Verk Web 😶

keyan commented:
No problem about releasing Verk Web, I've been running the `master` branch in production anyway!
